app/assets/javascripts/app/views/search_base_view.js

```diff
@@ -0,0 +1,141 @@
+app.views.SearchBase = app.views.Base.extend({
+  completeSetup: function(typeaheadElement){
```

Comment:
Wouldn't that conflicts with the `initialize` function redefined in the inheriting classes?